creator-node/src/middlewares.js

```diff
@@ -186,6 +187,118 @@ async function triggerSecondarySyncs (req) {
+    return
+  }
+  const wallet = req.session.wallet
+
+  try {
+    if (!req.session.nodeIsPrimary || !req.session.creatorNodeEndpoints || !Array.isArray(req.session.creatorNodeEndpoints)) {
+      req.logger.info('TODO')
+      return
+    }
+
+    let [primary, ...secondaries] = req.session.creatorNodeEndpoints
+    secondaries = secondaries.filter(secondary => (!!secondary && _isFQDN(secondary)))
+
+    if (primary !== config.get('creatorNodeEndpoint')) {
+      throw new Error('TODO')
+    }
+
+    const timeoutMs = enforceQuorum ? 20000 : 5000
+
+    /**
+     * Keep attempting to send syncRequest to secondary until it has synced or until timeoutMs
+     * TODO consider replacing with SnapbackSM.additionalSyncIsRequired code
+     */
+    const secondaryPromise = async function (secondaryUrl) {
+      const start = Date.now()
+      while (Date.now() - start < timeoutMs) {
+        try {
+          const secondaryClockStatusResp = await axios({
+            method: 'get',
+            baseURL: secondaryUrl,
+            url: `/users/clock_status/${wallet}`,
+            responseType: 'json',
+            timeout: 1000 // 1000ms = 1s
+          })
+          const { clockValue: secondaryClockVal } = secondaryClockStatusResp.data.data
+
+          // If secondary is synced, flip flag + return
+          if (secondaryClockVal >= primaryClockVal) {
+            return
+          }
+
+          // Send syncRequest to secondary; if immediate = true, submit POST request directly, else submit via SnapbackSM
+          if (immediate) {
+            await axios({
+              method: 'post',
+              baseURL: secondaryUrl,
+              url: '/sync',
+              data: {
+                wallet: [wallet],
+                creator_node_endpoint: primary,
+                immediate
+              },
+              timeout: timeoutMs * 2
+            })
+          } else {
+            await snapbackSM.enqueueSync({
+              userWallet: wallet,
+              secondaryEndpoint: secondaryUrl,
+              primaryEndpoint: primary,
+              syncType: SyncType.Manual
+            })
+
+            // Since enqueueSync is such a quick operation, give secondary a second to process it before next loop iteration
+            await utils.timeout(1000, false)
+          }
+        } catch (e) {
+          // do nothing and let while loop continue
+        }
+      }
+
+      // This condition will only be hit if the secondary has failed to sync within timeoutMs
+      throw new Error('TODO')
+    }
+    const secondaryPromises = secondaries.map(secondaryPromise)
+
+    // Race will throw if neither of secondaryPromises resolves within timeoutMs
+    try {
+      const timeoutPromise = new Promise((resolve, reject) => {
+        setTimeout(() => reject(new Error('Timeout')), timeoutMs)
+      })
+
+      await Promise.race([...secondaryPromises, timeoutPromise])
```

Comment:
I think we want promise any here

Reply:
gooood call, i actually thought about this and picked the wrong one lol

Reply:
fixed